clippy_lints/src/collapsible_if.rs

```diff
@@ -163,6 +164,54 @@ impl CollapsibleIf {
         }
     }
 
+    fn check_collapsible_if_nested_if_else(context: &LateContext<'_>, if_expr: &Expr<'_>) {
+        let ExprKind::If(cond, then, else_opt) = if_expr.kind else {
+            return;
+        };
+
+        let ExprKind::Block(then_block, _) = then.kind else {
+            return;
+        };
+
+        let Some(then_expr) = then_block.expr else {
+            return;
+        };
+
+        let ExprKind::If(inner_cond, inner_then, _) = then_expr.kind else {
+            return;
+        };
+
+        let Some(else_expr) = else_opt else {
+            return;
+        };
+
+        let mut spanless_eq = SpanlessEq::new(context);
+
+        if !spanless_eq.eq_expr(inner_then, else_expr) {
+            return;
+        }
```

Comment:
Please use a `if let` chain instead of doing 6 early returns, this is as clear and much shorter.